File: src/XmlSerializerCSDemo/Adres.cs
using System.Xml.Serialization;

namespace XmlSerializerCSDemo
{

    /// <summary>
    /// Adres Sýnýfý
    /// </summary>
    [XmlRoot]
    public class Adres
    {
        /// <summary>
        /// Gets or sets the identifier.
        /// </summary>
        /// <value>
        /// The identifier.
        /// </value>
        [XmlAttribute]
        public long Id { get; set; }

        /// <summary>
        /// Gets or sets the adres tipi.
        /// </summary>
        /// <value>
        /// The adres tipi.
        /// </value>
        [XmlAttribute]
        public AdresTipi AdresTipi { get; set; }

        /// <summary>
        /// Gets or sets the adres satiri1.
        /// </summary>
        /// <value>
        /// The adres satiri1.
        /// </value>
        [XmlElement]
        public string AdresSatiri1 { get; set; }

        /// <summary>
        /// Gets or sets the adres satiri2.
        /// </summary>
        /// <value>
        /// The adres satiri2.
        /// </value>
        [XmlElement]
        public string AdresSatiri2 { get; set; }

        /// <summary>
        /// Gets or sets the ilce.
        /// </summary>
        /// <value>
        /// The ilce.
        /// </value>
        [XmlElement]
        public string Ilce { get; set; }

        /// <summary>
        /// Gets or sets the il.
        /// </summary>
        /// <value>
        /// The il.
        /// </value>
        [XmlElement]
        public string Il { get; set; }
    }

}

File: src/XmlSerializerDemo/Adres.vb
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Converters
Imports System.Xml.Serialization

Namespace XmlSerializerDemo

    ''' <summary>
    ''' Adres Sınıfı
    ''' </summary>
    <Serializable, XmlRoot, JsonObject>
    Public Class Adres
        ''' <summary>
        ''' Gets or sets the identifier.
        ''' </summary>
        ''' <value>
        ''' The identifier.
        ''' </value>
        <XmlAttribute, JsonProperty("id")>
        Public Property Id() As Long

        ''' <summary>
        ''' Gets or sets the adres tipi.
        ''' </summary>
        ''' <value>
        ''' The adres tipi.
        ''' </value>
        <XmlAttribute, JsonProperty("adresTipi"), JsonConverter(GetType(StringEnumConverter))>
        Public Property AdresTipi() As AdresTipi

        ''' <summary>
        ''' Gets or sets the adres satiri1.
        ''' </summary>
        ''' <value>
        ''' The adres satiri1.
        ''' </value>
        <XmlElement, JsonProperty("adresSatiri1")>
        Public Property AdresSatiri1() As String

        ''' <summary>
        ''' Gets or sets the adres satiri2.
        ''' </summary>
        ''' <value>
        ''' The adres satiri2.
        ''' </value>
        <XmlElement, JsonProperty("adresSatiri2")>
        Public Property AdresSatiri2() As String

        ''' <summary>
        ''' Gets or sets the ilce.
        ''' </summary>
        ''' <value>
        ''' The ilce.
        ''' </value>
        <XmlElement, JsonProperty("ilce")>
        Public Property Ilce() As String

        ''' <summary>
        ''' Gets or sets the il.
        ''' </summary>
        ''' <value>
        ''' The il.
        ''' </value>
        <XmlElement, JsonProperty("il")>
        Public Property Il() As String
    End Class

End Namespace
